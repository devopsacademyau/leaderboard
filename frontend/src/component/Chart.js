import React, { Fragment } from "react"

const Chart = ({ chart, authors }) => (
  <Fragment>
    <h2>{chart.title}</h2>
    <table>
      <tbody>
        {chart.points.map((point, i) => {
          const a = authors[point.author]
          const aUrl = a ? a.url : ""
          return (
            <tr key={i}>
              <td>
                <a href={aUrl} target="_blank" rel="noopener noreferrer">
                  {point.author}
                </a>
              </td>
              <td>{point.count}</td>
            </tr>
          )
        })}
      </tbody>
    </table>
  </Fragment>
)

export default Chart
