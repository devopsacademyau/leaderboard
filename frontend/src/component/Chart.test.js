import React from "react"
import { Chart, Heading, getPrQuery, getAuthorQuery } from "./Chart"
import renderer from "react-test-renderer"

test("getPrQuery", () => {
  const prMergedQuery = getPrQuery("Merged or Open Pull Requests", "2019-01-01")
  expect(prMergedQuery).toBe("is:pr is:merged,open created:>2019-01-01")
  const prOpenQuery = getPrQuery("Merged Pull Requests", "2019-01-01")
  expect(prOpenQuery).toBe("is:pr is:merged created:>2019-01-01")
})

test("getAuthorQuery", () => {
  const authorQuery = getAuthorQuery(
    "Merged or Open Pull Requests",
    "my-author"
  )
  expect(authorQuery).toBe(" author:my-author")
})

test("matches Heading snapshot", () => {
  const tree = renderer.create(<Heading title="Code Reviews" />).toJSON()
  expect(tree).toMatchSnapshot()
})

test("matches Chart snapshot charts with unknown title", () => {
  const chart = {
    title: "My Chart",
    maxCount: 3,
    totalCount: 6,
    points: [
      { author: "a", count: "1" },
      { author: "b", count: "2" },
      { author: "c", count: "3" }
    ]
  }
  const authors = {
    a: {
      login: "a",
      url: "https://github.com/a",
      avatarURL: "https://avatars3.githubusercontent.com/u/22388715?v=4"
    },
    b: {
      login: "b",
      url: "https://github.com/b",
      avatarURL: "https://avatars0.githubusercontent.com/u/6849798?v=4"
    }
  }
  const config = {
    labelGlob: "lab.*",
    botName: "golangcibot",
    createdAfter: "2019-05-15T00:00:00Z",
    repository: {
      name: "academy",
      owner: "devopsacademyau",
      url: "https://github.com/devopsacademyau/academy"
    }
  }

  const tree = renderer
    .create(<Chart chart={chart} authors={authors} config={config} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
