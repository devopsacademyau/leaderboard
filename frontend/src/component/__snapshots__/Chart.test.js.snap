// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Chart snapshot charts with unknown title 1`] = `
Array [
  <h2
    className="f4 f3-ns fw6 pt4 pb2-ns"
  >
    <a
      className="link black-80 hover-blue underline-hover"
      href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15"
      rel="noopener noreferrer"
      target="_blank"
    >
      My Chart
    </a>
  </h2>,
  <table
    className="dt--fixed collapse f7 f6-ns"
  >
    <tbody>
      <tr
        className="w-100"
      >
        <td
          className="w2 pr1 pt1"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/a"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="avatar"
              className="h1 w1"
            />
          </a>
        </td>
        <td
          className="w4 pt1 truncate"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/a"
            rel="noopener noreferrer"
            target="_blank"
          >
            a
          </a>
        </td>
        <td
          className="w2 pt1"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15%20reviewed-by%3Aa"
            rel="noopener noreferrer"
            target="_blank"
          >
            1
          </a>
        </td>
        <td
          className="w-100"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15%20reviewed-by%3Aa"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              className="bg-blue dim"
              style={
                Object {
                  "height": "1rem",
                  "width": "33.33333333333333%",
                }
              }
            >
               
            </div>
          </a>
        </td>
      </tr>
      <tr
        className="w-100"
      >
        <td
          className="w2 pr1 pt1"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/b"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="avatar"
              className="h1 w1"
            />
          </a>
        </td>
        <td
          className="w4 pt1 truncate"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/b"
            rel="noopener noreferrer"
            target="_blank"
          >
            b
          </a>
        </td>
        <td
          className="w2 pt1"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15%20reviewed-by%3Ab"
            rel="noopener noreferrer"
            target="_blank"
          >
            2
          </a>
        </td>
        <td
          className="w-100"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15%20reviewed-by%3Ab"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              className="bg-blue dim"
              style={
                Object {
                  "height": "1rem",
                  "width": "66.66666666666666%",
                }
              }
            >
               
            </div>
          </a>
        </td>
      </tr>
      <tr
        className="w-100"
      >
        <td
          className="w2 pr1 pt1"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="avatar"
              className="h1 w1"
              src="missing-avatar-url"
            />
          </a>
        </td>
        <td
          className="w4 pt1 truncate"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href=""
            rel="noopener noreferrer"
            target="_blank"
          >
            c
          </a>
        </td>
        <td
          className="w2 pt1"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15%20reviewed-by%3Ac"
            rel="noopener noreferrer"
            target="_blank"
          >
            3
          </a>
        </td>
        <td
          className="w-100"
        >
          <a
            className="link black-80 hover-blue underline-hover"
            href="https://github.com/devopsacademyau/academy/pulls?q=is%3Apr%20created%3A%3E2019-05-15%20reviewed-by%3Ac"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              className="bg-blue dim"
              style={
                Object {
                  "height": "1rem",
                  "width": "100%",
                }
              }
            >
               
            </div>
          </a>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`matches Heading snapshot 1`] = `
<h2
  className="f4 f3-ns fw6 pt4 pb2-ns"
>
  Code Reviews
</h2>
`;
